lib/emis/responses/response.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+require 'ostruct'
+
+module EMIS
+  module Responses
+    class Response
+      def initialize(raw_response)
+        @root = raw_response.body
+      end
+
+      def items
+        locate(item_tag_name).map do |el|
+          build_item(el)
+        end
+      end
+
+      def ok?
+        locate_one('essResponseCode')&.nodes&.first == 'Success'
+      end
+
+      def error?
+        locate_one('essResponseCode')&.nodes&.first == 'ERROR'
+      end
+
+      def empty?
+        locate_one('essResponseCode')&.nodes&.first.nil?
+      end
+
+      def locate(tag_without_namespace, el = @root)
+        find_all_elements_by_tag_name(tag_without_namespace, el, skip_el: true)
+      end
+
+      protected
+
+      def build_item(el, schema: item_schema)
+        result_hash = {}.tap do |result|
+          schema.each do |tag, data|
+            field_name = data[:rename] || tag.snakecase
+            if data[:schema]
+              tags = locate(tag, el)
+              result[field_name] = tags.map { |t| build_item(t, schema: data[:schema]) }
+            else
+              build_item_value(el, tag, data, field_name, result)
+            end
+          end
+        end
+
+        OpenStruct.new(result_hash)
```

Comment:
also not a big fan of using OpenStruct. I know Virtus is currently in search of a new maintainer and there are a bunch of alternatives to a coercion library like it, but I think it might be better than using open struct seeing as it is the pattern we've already used elsewhere.

In other words, we can use a parser middleware to get the data into a format where it can be deserialized / marshalled into a Virtus object. And Virtus has many coercion types that can then be used to coerce types. In addition, one can easily create his or her own coercion types, for example: Common::UTCTime